packages/go/trynoports/app_update.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+// Update part of Elm architecture for app
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/charmbracelet/bubbles/viewport"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/charmbracelet/log"
+)
+
+func (m appState) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var (
+		cmd  tea.Cmd
+		cmds []tea.Cmd
+	)
+
+	// Handle terminal / input events first
+	switch msg := msg.(type) {
+	case ProgramMsg:
+		m.program = msg.program
+	case CmdDoneMsg:
+		m.viewport.isRunning = false
+	case ViewportContentMsg:
+		m.viewport.content += msg.content
+		m.viewport.model.SetContent(m.viewport.content)
+		m.viewport.model.GotoBottom()
+		if useHighPerformanceRenderer {
+			viewport.Sync(m.viewport.model)
+		}
+	case tea.WindowSizeMsg:
+		m.width = msg.Width
+		m.height = msg.Height
+
+		// We have an initial window size now
+		// Now it is safe to initialize the viewport
+		if !m.viewport.isReady {
+			// We will resize this to the correct size during the ResizeComponents call
+			m.viewport.model = viewport.New(1, 1)
+			m.viewport.model.SetContent(welcomeMessageContent)
+			m.viewport.model.HighPerformanceRendering = useHighPerformanceRenderer
+			m.viewport.model.KeyMap.Down.SetEnabled(false)
+			m.viewport.model.KeyMap.Up.SetEnabled(false)
+			m.viewport.isReady = true
+		}
```

Comment:
Rest of the viewport initialization, had to wait for the program to report an initial window size.